packages/components/src/internal/components/samples/actions.ts

```diff
@@ -396,77 +394,85 @@ function getSamplesIdsNotFound(queryName: string, orderedIds: string[]): Promise
     });
 }
 
-export function getFindSamplesByIdData(): Promise<{ queryName: string; missingIds?: { [key: string]: string[] } }> {
+export function getFindSamplesByIdData(sessionKey: string): Promise<{ queryName: string; ids: string[], missingIds?: { [key: string]: string[] } }> {
     return new Promise((resolve, reject) => {
-        const ids = JSON.parse(sessionStorage.getItem(FIND_IDS_SESSION_STORAGE_KEY));
-        if (ids) {
+        Ajax.request({
+            url: ActionURL.buildURL('experiment', 'saveOrderedSamplesQuery.api'),
+            method: 'POST',
+            jsonData: {
+                sessionKey
+            },
+            success: Utils.getCallbackWrapper(response => {
+                if (response.success) {
+                    const { queryName, ids } = response.data;
+                    getSamplesIdsNotFound(queryName, ids)
+                        .then(notFound => {
+                            const missingIds = {
+                                [UNIQUE_ID_FIND_FIELD.label]: notFound
+                                    .filter(id => id.startsWith(UNIQUE_ID_FIND_FIELD.storageKeyPrefix))
+                                    .map(id => id.substring(UNIQUE_ID_FIND_FIELD.storageKeyPrefix.length)),
+                                [SAMPLE_ID_FIND_FIELD.label]: notFound
+                                    .filter(id => id.startsWith(SAMPLE_ID_FIND_FIELD.storageKeyPrefix))
+                                    .map(id => id.substring(SAMPLE_ID_FIND_FIELD.storageKeyPrefix.length)),
+                            };
+                            resolve({
+                                queryName,
+                                ids,
+                                missingIds,
+                            });
+                        })
+                        .catch(reason => {
+                            console.error('Problem retrieving data about samples not found', reason);
+                            resolve({
+                                queryName,
+                                ids
+                            });
+                        });
+                } else {
+                    console.error('Unable to create session query');
+                    reject('There was a problem creating the query for the samples. Your session may have expired.');
+                }
+            }),
+            failure: Utils.getCallbackWrapper(error => {
+                console.error('There was a problem creating the query for the samples.', error);
+                reject("There was a problem creating the query for the samples. Please try again using the 'Find Samples' option from the Search menu.");
```

Comment:
Should any of these error messages be updated given that this is a "get" action/function instead of the one "creating" the query?

Reply:
It is actually creating the query, but users may not need to know that.  Could just be "There was a problem retrieving the samples." for the user message.